### PR TITLE
adds support for using connection=netconf

### PR DESCRIPTION
**Pull request by privateip**
This change updates the module to provide support for using
connection=netconf instead of connection=local.  If connection=netconf
is used, then the various connection arguments will be silently ignored.

### CODE REVIEW

**Comment by lpenz** [comment]
Looks great!
I'm not being able to test it, though. After changing some things to move the code forward, I end up getting something like
    AnsibleError: connection=netconf is not supported on <x>
...where <x> is whatever I set in ansible_network_os.

I understand that this is WIP, but let me know if I should not be seeing this.

**Comment by privateip** [comment]
This PR is dependent on #33671

**Comment by privateip** [comment]
@lpenz thats strange I cannot reproduce the error you are referencing.  There is a bug that is dependent on 33671 being merged but that is a different issue.  once 33671 is merged, could you please retest?

**Comment by lpenz** [comment]
Will do, thanks!

**Comment by privateip** [comment]
@lpenz do you have a chance to test this PR?

File: lib/ansible/modules/network/netconf/netconf_config.py [hunk 1]
```diff
@@ -214,68 +244,82 @@ def main():
...
-        port=module.params['port'],
-        hostkey_verify=module.params['hostkey_verify'],
-        allow_agent=module.params['allow_agent'],
-        look_for_keys=module.params['look_for_keys'],
-        username=module.params['username'],
-        password=module.params['password'],
-    )
+    if not local_connection:
+        m = Connection(module._socket_path)
+        capabilities = module.from_json(m.get_capabilities())
```
Comment (lpenz) [inline lib/ansible/modules/network/netconf/netconf_config.py:245]: I have been hitting an issue here, where `m` ends up being a module_utils.connection.Connection without a `get_capabilities` method. I'm using `ansible_network_os=default` to access an openyuma server. That works without using `ansible_connection=netconf`

**Comment by privateip** [comment]
@lpenz that helps, thanks.  I see where the problem is and will have a fix shortly

**Comment by privateip** [comment]
@lpenz I finally got around to putting that fix in.  This should now work without having to specify `ansible_network_os` variable.

**Comment by lpenz** [comment]
I've tested it, and it's now working.
Thanks for doing this! It makes ansible look a lot more consistent when dealing with network devices.

**Comment by lpenz** [comment]
I tried this out, and it is looking better, though I got another problem that will demand more debugging time.
(it's complaining that ncclient is not installed)

**Review by ganeshrn — APPROVED** [review]

**Pull request merged**